src/select.ts

```diff
@@ -0,0 +1,129 @@
+  try {
+    await SelectOptionsList(labelId).perform((element) => {
+      const popover = element.parentElement?.parentElement;
+      if (popover?.getAttribute("role") == "presentation" && popover.classList.contains("MuiPopover-root")) {
+        const popoverTrap = popover.firstElementChild;
+        if (isHTMLElement(popoverTrap)) popoverTrap.click();
+      }
+    });
+  } catch (_) {
+    /* noop */
+  }
+}
+
+function getInputLabel(element: HTMLElement) {
+  const input = element.nextElementSibling;
+  return isHTMLElement(input, "Input") ? input.labels?.[0] : undefined;
+}
+
+async function toggleOption(
+  interactor: Interactor<HTMLElement, any>,
+  value: string,
+  { clearSelection = false }: { clearSelection?: boolean } = {}
+) {
+  let labelId = "";
+  let selected: string[] = [];
+  await interactor.perform((element) => (labelId = getInputLabel(element)?.id ?? ""));
+  await interactor.perform((element) => dispatchMouseDown(element));
+  await delay(100);
+  try {
+    if (clearSelection) {
+      await SelectOptionsList(labelId).perform(
+        (element) =>
+          (selected = Array.from(element.querySelectorAll("li"))
+            .filter((option) => option.getAttribute("aria-selected") == "true")
+            .map((option) => option.innerText))
+      );
+      for (const option of selected) {
+        await SelectOptionsList(labelId).find(Option(option)).choose();
+      }
+    }
+    await SelectOptionsList(labelId).find(Option(value)).choose();
+  } finally {
+    await closeOptionsList(labelId);
+  }
+}
+
+const BaseSelect = createInteractor<HTMLElement>("MUI BaseSelect")
+  .selector(".MuiSelect-root")
+  .locator((element) => {
+    const label = getInputLabel(element);
+    return label ? innerText(label) : "";
+  })
+  .filters({
+    id: (element) => element.id,
+    className: (element) => Array.from(element.parentElement?.classList ?? []),
+    valid: (element) => !element.parentElement?.classList.contains("Mui-error"),
+    required: (element) => element.parentElement?.hasAttribute("required"),
+    disabled: {
+      apply: (element) => element.getAttribute("aria-disabled") == "true",
+      default: false,
+    },
+  });
+
+export const Select = BaseSelect.extend("MUI Select")
+  .filters({
+    value: (element) => element.innerText,
+  })
+  .actions({
+    choose: async (interactor, value: string) => {
+      try {
+        await interactor.has({ value });
+      } catch (_) {
+        await toggleOption(interactor, value, { clearSelection: true });
+      }
+    },
+  });
+
+export const MultiSelect = BaseSelect.extend("MUI MultiSelect")
+  .filters({
+    values: (element) =>
+      Array.from(element.querySelectorAll(".MuiChip-root > .MuiChip-label") ?? [])
+        .map((chip) => (isHTMLElement(chip) ? chip.innerText : null))
+        .filter(isDefined),
+  })
+  .actions({
+    choose: async (interactor, value: string) => {
+      try {
+        await interactor.has({ values: [value] });
```

Comment:
We don't want to wait for the entire timeout period to proceed if the selection is already there that we need. Let's extract the values filter into a stand-alone function that we can invoke from `interactor.peform`.